src/EditorFeatures/CSharpTest/MoveStaticMembers/CSharpMoveStaticMembersTests.cs

```diff
@@ -0,0 +1,636 @@
+            var expectedResult = @"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""CSAssembly1"" CommonReferences=""true"">
+        <Document FilePath=""Class1.cs"">
+namespace TestNs1
+{
+    public class Class1
+    {
+    }
+}
+        </Document>
+        <Document FilePath=""Class1Helpers.cs"">namespace TestNs1
+{
+    static class Class1Helpers
+    {
+        public static int TestField = 1;
+    }
+}</Document>
+    </Project>
+</Workspace>";
+            await TestWithDialogAsync(initialMarkup,
+                expectedResult,
+                selectedDestinationName,
+                selectedMembers,
+                newFileName).ConfigureAwait(false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveStaticMembers)]
+        public async Task TestMoveProperty()
+        {
+            var initialMarkup = @"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""CSAssembly1"" CommonReferences=""true"">
+        <Document FilePath=""Class1.cs"">
+namespace TestNs1
+{
+    public class Class1
+    {
+        public static int Test[||]Property { get; set; }
+    }
+}
+        </Document>
+    </Project>
+</Workspace>";
+            var selectedDestinationName = "Class1Helpers";
+            var newFileName = "Class1Helpers.cs";
+            var selectedMembers = ImmutableArray.Create("TestProperty");
+            var expectedResult = @"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""CSAssembly1"" CommonReferences=""true"">
+        <Document FilePath=""Class1.cs"">
+namespace TestNs1
+{
+    public class Class1
+    {
+    }
+}
+        </Document>
+        <Document FilePath=""Class1Helpers.cs"">namespace TestNs1
+{
+    static class Class1Helpers
+    {
+        public static int TestProperty { get; set; }
+    }
+}</Document>
+    </Project>
+</Workspace>";
+            await TestWithDialogAsync(initialMarkup,
+                expectedResult,
+                selectedDestinationName,
+                selectedMembers,
+                newFileName).ConfigureAwait(false);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsMoveStaticMembers)]
+        public async Task TestMoveEvent()
+        {
+            var initialMarkup = @"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""CSAssembly1"" CommonReferences=""true"">
+        <Document FilePath=""Class1.cs"">
+using System;
+
+namespace TestNs1
+{
+    public class Class1
+    {
+        public static event EventHandler Test[||]Event;
+    }
+}
+        </Document>
+    </Project>
+</Workspace>";
+            var selectedDestinationName = "Class1Helpers";
+            var newFileName = "Class1Helpers.cs";
+            var selectedMembers = ImmutableArray.Create("TestEvent");
+            var expectedResult = @"
+<Workspace>
+    <Project Language=""C#"" AssemblyName=""CSAssembly1"" CommonReferences=""true"">
+        <Document FilePath=""Class1.cs"">
+using System;
```

Comment:
This is interesting. Not for this PR but we may want to consider removing using directives that become unnecessary once the members are moved.

Reply:
My reasoning for not doing this in the PullMemberUp (which is currently handling the importing) is that users may want to keep certain imports in some/all their files, even if they aren't used. I think this would have a similar argument against.